plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/IcebergMetadata.java

```diff
@@ -1150,6 +1157,22 @@ public ConnectorInsertTableHandle beginRefreshMaterializedView(ConnectorSession
         return catalog.listMaterializedViews(session, schemaName);
     }
 
+    @Override
+    public Map<SchemaTableName, ConnectorMaterializedViewDefinition> getMaterializedViews(ConnectorSession session, Optional<String> schemaName)
```

Comment:
Should this change just go into the default implementation in `ConnectorMetadata`?

Reply:
Iceberg has a challenge because it keeps state between metastore and filesystem, and cannot rely on observing consistent state. I would hope most connectors don't have problems like that. Or, may be able to better exception-control this (here i catch `RuntimeException` which is too broad in a general case).